web/src/components/dropdowns/group-dropdown.tsx

```diff
@@ -73,26 +73,19 @@ export const GroupDropdown: React.FC<{
         </DropdownToggle>
       }
       isOpen={groupDropdownOpen}
-      dropdownItems={getAvailableGroups(scope)
-        .filter(
-          g =>
-            (allowMultiCluster ||
-              ![
-                TopologyGroupTypes.CLUSTERS,
-                TopologyGroupTypes.CLUSTERS_HOSTS,
-                TopologyGroupTypes.CLUSTERS_NAMESPACES,
-                TopologyGroupTypes.CLUSTERS_OWNERS,
-                TopologyGroupTypes.CLUSTERS_ZONES
-              ].includes(g)) &&
-            (allowZone ||
-              ![
-                TopologyGroupTypes.ZONES,
-                TopologyGroupTypes.ZONES_HOSTS,
-                TopologyGroupTypes.ZONES_NAMESPACES,
-                TopologyGroupTypes.ZONES_OWNERS,
-                TopologyGroupTypes.CLUSTERS_ZONES
-              ].includes(g))
-        )
+      dropdownItems={getGroupsForScope(scope)
+        .filter(g => {
+          if (
+            (!allowedScopes.includes('cluster') && g.includes(TopologyGroupTypes.CLUSTERS)) ||
+            (!allowedScopes.includes('zone') && g.includes(TopologyGroupTypes.ZONES)) ||
+            (!allowedScopes.includes('host') && g.includes(TopologyGroupTypes.HOSTS)) ||
+            (!allowedScopes.includes('namespace') && g.includes(TopologyGroupTypes.NAMESPACES)) ||
+            (!allowedScopes.includes('owner') && g.includes(TopologyGroupTypes.OWNERS))
+          ) {
+            return false;
```

Comment:
I am no very convienced this producing the same logic as prev code ?

Reply:
the logic changes but the result should be the same. The goal is to not show, for instance, cluster-related groupings (e.g. "clusters" or "clusters+namespaces") when multi-cluster feature is disabled.

Previously we were filtering out group names if they were in a list ["clusters", "clusters+namespaces", etc.]
Now we filter out group names if they contain, for instance, the string "clusters". This should work because all cluster-related groups contain the substring "clusters".

I guess I can add a unit test for the proof :-)